presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -252,6 +252,9 @@ default ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTabl
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support deletes");
     }
 
+    default void beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
Why is `ConnectorTableLayoutHandle` optional? There should always be a layout.

Since we always have a layout, I think both the table and column handles are redundant, since both of those are provided by the `getTableLayouts()` call which returns the layout.